lib/dune

```diff
@@ -1,6 +1,6 @@
 (library
  (name        httpaf)
  (public_name httpaf)
- (libraries
-   angstrom faraday bigstringaf)
+ (libraries angstrom faraday bigstringaf base sexplib)
```

Comment:
Do we really need `sexplib` here?

Reply:
Yes, for the `[@deriving sexp]` on the types.  However, I don't think we need the `base` dependency, so I've removed it.

Reply:
Hmmhmm, but do we really need `[@deriving sexp]`?

It is of very limited interest. As far as I can see, there is no internal use for the deriver in the project - so it responds to an external demand (I suppose, in this case, JaneStreet). However, there is no demand (in the issues) for such a deriver and I'm not personally in favour of using it.

Reply:
> As far as I can see, there is no internal use for the deriver in the project - so it responds to an external demand

They're used in some `expect_test`s that I'm still in the process of upstreaming.

> so it responds to an external demand (I suppose, in this case, JaneStreet).

All of the people who have contributed to httpaf currently work at Jane Street, so the internal / external divide isn't totally clear.  I also won't pretend to speak on their behalf, I'm just doing the work of pushing the changes that they already made.

> Hmmhmm, but do we really need [@deriving sexp]?

The dependency is very small, and if you have code that uses the sexp deriver, then having the serializers and deserializers directly in the library means that you don't have to re-declare the types to add the derivers and keep them in sync.

Reply:
> All of the people who have contributed to httpaf currently work at Jane Street

That's a little simplistic...

>  I also won't pretend to speak on their behalf, I'm just doing the work of pushing the changes that they already made.

There are several questions here:
1) how do maintainers/authors view `httpaf` and its community.
2) JaneStreet surely has an internal use of `httpaf` but it seems that not all of it has been exported here (furthermore, #159 or #203 exist and do not add the deriver)
3) `httpaf` is used by other projects (notably dream) for which adding a dependency (even a small one) has real implications.

For example, as far as Mirage is concerned, we've already had problems with `sexplib0` (which we've of course fixed).

> The dependency is very small, and if you have code that uses the sexp deriver, then having the serializers and deserializers directly in the library means that you don't have to re-declare the types to add the derivers and keep them in sync.

This remains a very specific use. Once again, no one in the community has asked for such a deriver, even if it's only in the issues. It should also be understood that it can be frustrating to have a dependency imposed on you as a long-standing user (and contributor in a way - as far as `faraday`, `angstrom` or `bigstringaf` are concerned) of `httpaf` for a reason that is as obscure as it is internal to JaneStreet.